### PR TITLE
pkg/cvo/metrics: Add a cluster_version_capability metric

### PR DESCRIPTION
**Pull request by wking**
Fleet capability choices should make their way up to Red Hat via Insights uploads.  But [some clusters have Telemetry enabled but Insights disabled][1].  For these clusters, we'll want this data in Telemetry, and getting it into metrics (this commit) is the first step.  After this lands, we'll need to update the monitoring operator like [this][2] to get these shipped up off of cluster.

[1]: https://github.com/openshift/enhancements/pull/922#discussion_r742287363
[2]: https://github.com/openshift/cluster-monitoring-operator/pull/1477

### CODE REVIEW

**Comment by wking** [comment]
PromeCIeus for [the e2e-operator run][1] has:

```
cluster_version_capability{endpoint="metrics", instance="10.0.0.7:9099", job="cluster-version-operator", name="baremetal", namespace="openshift-cluster-version", pod="cluster-version-operator-86f88db7f6-xbw5n", service="cluster-version-operator"} 
 1
cluster_version_capability{endpoint="metrics", instance="10.0.0.7:9099", job="cluster-version-operator", name="openshift-samples", namespace="openshift-cluster-version", pod="cluster-version-operator-86f88db7f6-xbw5n", service="cluster-version-operator"}  1
```

So hurray :)

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/755/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-operator/1506518234898108416

**Comment by wking** [comment]
[Azure install failure][1] is already tracked in an installer bug, and I'm confident that there's no update-specific impact here.

/override ci/prow/e2e-agnostic-upgrade

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/755/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade/1506665607297241088

**Comment by wking** [comment]
Manually labeling per [Jack's earlier comment][1], because Prow and GitHub are fighting.

[1]: https://github.com/openshift/cluster-version-operator/pull/755#issuecomment-1076473892

File: pkg/cvo/metrics.go [hunk 1]
```diff
@@ -432,6 +438,21 @@ func (m *operatorMetrics) Collect(ch chan<- prometheus.Metric) {
...
 		}
 
+		enabledCapabilities := make(map[configv1.ClusterVersionCapability]struct{}, len(cv.Status.Capabilities.EnabledCapabilities))
+		for _, capability := range cv.Status.Capabilities.EnabledCapabilities {
+			g := m.capability.WithLabelValues(string(capability))
+			g.Set(float64(1))
+			ch <- g
+			enabledCapabilities[capability] = struct{}{}
+		}
+		for _, capability := range cv.Status.Capabilities.KnownCapabilities {
```
Comment (LalatenduMohanty) [inline pkg/cvo/metrics.go:448]: I am not clear on why we need to send known capabilities.

Comment (wking) [reply pkg/cvo/metrics.go:448]: Two use cases:

* conditional updates using local PromQL can distinguish between "yeah, that cap is not enabled here" (value 0) and "err, something's broken with scraping this metrics" (no match)
* Telemetry can gauge the install-fraction for a capability (`countOfClustersWithCapEnabled / countOfClustersWithCapKnown`).  with `count by (name) (cluster_version_capability == 1) / count by (name) (cluster_version_capability)`.  If `cluster_version_capability` did not include known-but-not-enabled caps, you'd need some added filtering to get at `countOfClustersWithCapKnown`.

**Pull request merged**